src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs

```diff
@@ -59,7 +59,9 @@ public void CompileTempPE(string pszOutputFileName, int sourceCount, string[] fi
                     .WithXmlReferenceResolver(XmlFileResolver.Default)
                     .WithMetadataReferenceResolver(metadataResolver));
 
-            compilation.Emit(pszOutputFileName);
+            var result = compilation.Emit(pszOutputFileName);
+
+            Contract.ThrowIfFalse(result.Success);
```

Comment:
I'm also concerned about this - what did the old TempPE Compile do on compilation error?